src/main/java/games/strategy/engine/lobby/server/db/UserDao.java

```diff
@@ -9,10 +9,16 @@
   boolean doesUserExist(String userName);
 
   void updateUser(DBUser user, HashedPassword password);
+  
+  void updateUser(DBUser user, String password);
```

Comment:
I would suggest we keep the `HashedPassword`, it was introduced recently to avoid 'Stringly' types. It was of help too, we had two String params side by side that I did have inverted, the strong typed `HashedPassword` fixed the error. So the value object is there to avoid passing around passwords as `String` types. An analogy in my mind is if someone were to use `double` objects when they were passing around `ints`. Sure `double` works, but the real data type is a subset, and semantically and logically has a slightly different meaning.

Though regardless, the update here still has issues, we should not have these two APIs living next to each other.

> If we no longer require the client to send hashed passwords, we don't need to check if they are valid hashes

In that case `HashedPassword` might be renamed to `ClientPassword`. Once we have a lot of complexity removed, then IMO we can look at removing value objects in favor of weaker typing. For now, given the complexity, upgrading type strength, eg: strings to enums or value objects is a good step to help us deal with the issues in this code base.